# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../squid/squid.3.1.18-build.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -up squid-3.1.18/src/store.cc.fix squid-3.1.18/src/store.cc
--- squid-3.1.18/src/store.cc.fix	2012-01-23 14:46:39.000000000 +0100
+++ squid-3.1.18/src/store.cc	2012-01-23 14:47:17.000000000 +0100
@@ -368,7 +368,7 @@ StoreEntry::StoreEntry(const char *aUrl,
 
 #if USE_ADAPTATION
 void
-StoreEntry::deferProducer(const AsyncCall::Pointer &producer)
+StoreEntry::deferProducer(AsyncCall::Pointer &producer)
 {
     if (!deferredProducer)
         deferredProducer = producer;
diff -up squid-3.1.18/src/Store.h.fix squid-3.1.18/src/Store.h
--- squid-3.1.18/src/Store.h.fix	2012-01-23 14:48:23.000000000 +0100
+++ squid-3.1.18/src/Store.h	2012-01-23 14:48:31.000000000 +0100
@@ -191,7 +191,7 @@ public:
 
 #if USE_ADAPTATION
     /// call back producer when more buffer space is available
-    void deferProducer(const AsyncCall::Pointer &producer);
+    void deferProducer(AsyncCall::Pointer &producer);
     /// calls back producer registered with deferProducer
     void kickProducer();
 #endif
